library/std/src/sys/windows/c.rs

```diff
@@ -511,6 +513,8 @@ pub struct REPARSE_DATA_BUFFER {
     pub rest: (),
```

Comment:
Does this really need to end with a zst? Given the issues, it might be worth exploring other options. The usual way to do this in msvc is to end with `[u8; 1]` (or similar). Or it could be a union of supported reparse types. Or it could be both, which is the [canonical definition](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/ns-ntifs-_reparse_data_buffer).

An alternative is to (ab)use the fact that the `REPARSE_DATA_BUFFER` header is 64bits on all supported platforms. Which is convenient for aligning the data (and is likely no accident).

Reply:
I think it's not great for it to end in a ZST, yeah.

Reply:
This wouldn't really change the situation with UB that much though -- we still couldn't return a `&REPARSE_DATA_BUFFER` without narrowing the provenance such that it'd be UB to access `PathBuffer`.

Reply:
Yeah, that's true. I'm not minded to block this PR on removing the zst but I do think we should be aiming to do so, and in general leaning toward not using them in Windows ffi so as to side-step #81996 completely. Especially as the Windows API doesn't really use them.

Reply:
I think that one would be fine, the issue isn't with all ZSTs, so much as zero-sized arrays like `[T; 0]` (since in a struct `T foo[0];` is the same as `T *foo;` on MSVC).